File: android/app/src/main/java/com/propertyfinder1/util/Interval.java
package com.propertyfinder1.util;

public class Interval {
    private int start;
    private int end;

    public Interval(int start, int end) {
        this.start = start;
        this.end = end;
    }

    public int getStart() {
        return start;
    }

    public int getEnd() {
        return end;
    }

    public void setStart(int start) {
        this.start = start;
    }

    public void setEnd(int end) {
        this.end = end;
    }

    @Override
    public boolean equals(Object obj) {
        Interval interval = (Interval)obj;
        return (this.start == interval.getStart() && this.end == interval.getEnd());
    }

    @Override
    public String toString() {
        return this.start+"-"+this.end;
    }

    public boolean contains(Interval interval) {
        return (this.start <= interval.getStart() && this.end >= interval.getEnd());
    }

    public boolean intersects(Interval interval) {
        return !(this.start >= interval.getEnd() || this.end <= interval.getStart());
    }

    public int compareTo(Interval interval) {
        if (this.start < interval.getStart()) {
            return -1;
        } else if (this.start > interval.getStart()) {
            return 1;
        } else {
            if (this.end == interval.getEnd()) {
                return 0;
            } else {
                return this.end < interval.getEnd() ? -1 : 1;
            }
        }
    }
}

File: ios/Interval.swift
//
//  Interval.swift
//  PropertyFinder1
//
//  Created by K Arun Kumar Reddy on 19/12/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

class Interval: Equatable, CustomStringConvertible {
    var description: String
    private var start: Int;
    private var end: Int;
    
    init(start: Int, end: Int) {
        self.start = start;
        self.end = end;
        self.description = start.description + "-" + end.description;
    }
    
    func getStart() -> Int {
        return self.start;
    }
    
    func getEnd() -> Int {
        return self.end;
    }
    
    func setStart(start: Int) {
        self.start = start;
    }
    
    func setEnd(end: Int) {
        self.end = end;
    }
    
    static func ==(a: Interval,b: Interval) -> Bool {
        return (a.getStart()==b.getStart() && a.getEnd()==b.getEnd());
    }
    
    func contains(interval: Interval) -> Bool {
        return (self.start <= interval.getStart() && self.end >= interval.getEnd());
    }
    
    func intersects(interval: Interval) -> Bool {
        return !(self.start >= interval.getEnd() || self.end <= interval.getStart());
    }
    
    func compareTo(interval: Interval) -> Int {
        if (self.start < interval.getStart()) {
            return -1;
        } else if (self.start > interval.getStart()) {
            return 1;
        } else {
            if (self.end == interval.getEnd()) {
                return 0;
            } else {
                return self.end < interval.getEnd() ? -1 : 1;
            }
        }
    }
    
}
